qualtran/bloqs/arithmetic/multiplication.py

```diff
@@ -41,7 +41,9 @@ def short_name(self) -> str:
 
     @property
     def signature(self) -> 'Signature':
-        return Signature.build(a=self.a_bitsize, b=self.b_bitsize, result=self.result_bitsize)
+        return Signature.build_from_dtypes(
+            a=QInt(self.a_bitsize), b=QInt(self.b_bitsize), result=QInt(self.result_bitsize)
+        )
```

Comment:
`PlusEqualProduct` is used to add into the phase gradient register (the `result` here) so the register types should be `QFxp` (I personally think most of the arithmetic gates should support `QFxp` but we can defer to later).

Reply:
Done.